Android: Add 16 KB page alignment support for Android 15+

### Problem
For Godot 4.5+, Rapier 2D does not support building from scratch. I tried to build it directly to fix the 16KB page alignment issue for Android, but some libraries need updates to work with the 4.5 version.

### Details
- **Godot Version**: 4.5
- **Current Issue**: `libgodot_rapier.android.aarch-64-linux-android.so` has 4KB page alignment
- **Required**: 16KB page alignment for Android 15+ compatibility

Thanks for the quick response! Let me clarify:

## The Two Issues We're Facing

### Issue 1: 16KB Page Alignment Requirement
- Google Play now requires all native libraries (.so files) to have 16KB page alignment for Android 15+ devices
- Current Rapier binaries (e.g., `libgodot_rapier.android.aarch-64-linux-android.so`) are compiled with 4KB alignment
- This causes apps to fail on newer Android devices with 16KB page sizes

### Issue 2: Can't Rebuild from Source with Godot 4.5
When attempting to rebuild Rapier with 16KB alignment for Godot 4.5, compilation fails with these errors:
error[E0407]: method set_collision_layer is not a member of trait IPhysicsDirectBodyState2DExtension
error[E0407]: method get_collision_layer is not a member of trait IPhysicsDirectBodyState2DExtension
error[E0407]: method set_collision_mask is not a member of trait IPhysicsDirectBodyState2DExtension
error[E0407]: method get_collision_mask is not a member of trait IPhysicsDirectBodyState2DExtension

--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -11,15 +11,25 @@ rustflags = [
   "-C", "link-args=-sDISABLE_EXCEPTION_CATCHING=1",
   "-Z", "link-native-libraries=no",
 ]
+
 [target.armv7-linux-androideabi]
 linker = "armv7a-linux-androideabi23-clang"
+rustflags = ["-C", "link-arg=-Wl,-z,max-page-size=16384"]
+
 [target.aarch64-linux-android]
 linker = "aarch64-linux-android23-clang"
+rustflags = ["-C", "link-arg=-Wl,-z,max-page-size=16384"]
+
 [target.x86_64-linux-android]
 linker = "x86_64-linux-android23-clang"
+rustflags = ["-C", "link-arg=-Wl,-z,max-page-size=16384"]
+
 [target.i686-linux-android]
 linker = "i686-linux-android23-clang"
+rustflags = ["-C", "link-arg=-Wl,-z,max-page-size=16384"]
+
 [target.aarch64-unknown-linux-gnu]
 linker = "aarch64-linux-gnu-gcc"
+
 [target.armv7-unknown-linux-gnueabihf]
 linker = "arm-linux-gnueabihf-gcc"

